### DISCUSSION TITLE
Parts: Copy-paste image support

### DISCUSSION THREAD

**Discussion by Petrox** [inventree/InvenTree#5488] [Ideas]
Low priority productivity enhancement idea:

The dialog to upload images
![image](https://github.com/inventree/InvenTree/assets/3036509/012bd9fd-bd86-4069-ae88-75f2a7bd05bb)
could support "CTRL-V" or "Paste from clipboard", after the user selects an image on the supplier website -> Right click -> Copy image.

I know it's not trivial to implement, but these efficiency improvements help a lot when there are hundreds of items to fulfil.

Additionally an "IMAGE URL" textbox could serve a similar purpose, but is limited to publicly online accessible image sources.

**Comment by Petrox** [comment]
(the image included in the text was Windows-Shift-F -> copy -> pasted into the text, since github, trello, jira, confluence, gitlab all support this productivity enhancement and we all prefer to avoid save-upload-pickthefolder-picktheimage processes.

**Comment by matmair** [comment]
You can already post a remote URL - as it is a security risk you need to enable the feature in the settings.

**Comment by wolflu05** [comment]
I like the idea of pasting images from the clipboard, often have screenshots saved in the clipboard and always have to save, pick and delete them. I think this should be handled directly in the new pui sometime if there is a generic file upload component.

**Reply by SchrodingersGat** [reply]
+1 for implementing this in the new UI. I am certain that this would be trivial with the new framework, potentially even drag-and-drop too.

**Comment by Petrox** [comment]
Still, this is low prio feature, but these things can help a user like one
system and hate another, so I would say these are important little things.

Definitely scheduled after the UI overhaul.

**Comment by Petrox** [comment]
This feature would be very-very helpful when creating parts from a cad
drawing by hand and each subpart can get a screen-paste easily in seconds.
